src/components/scripts/terminal.ts

```diff
@@ -1,191 +1,225 @@
-        else await typedCommand.function(args);
-        if (typedCommand != null) {
-            if (typedCommand.name == "access") {
-                scrollToLink();
-            }
-            else {
-                scrollPage();
-            }
-        } 
+  // Divide the command into the name and its arguments
+  const args = command.split(' ');
+  const typedCommand = commands.find(
+    (cmd: {name: string}) => cmd.name === args[0]
+  );
+  args.shift();
+
+  // TODO: Get rid of the old manual functionality. I havent figured out a good way to implement it yet
+  // Execute the command or return error message. Numbers for menu options do not get processed here.
+  if (isNaN(parseInt(command))) {
+    if (inDatabase || inManual || inMenu) {
+      console.log(command);
+      if (command === 'manual') {
+        inManual = false;
+        inMenu = false;
+        inDatabase = false;
+        command = '';
+        appendPrompt();
+        return;
+      }
+      termwindow.append(
+        span(
+          'status-fail',
+          "Not a supported menu option. To use commands, exit the menu by typing 'manual'\n"
+        )
+      );
+    } else if (typedCommand == null)
+      termwindow.append(span('status-fail', `Command not found: ${command}\n`));
+    else await typedCommand.function(args);
+    if (typedCommand != null) {
+      if (typedCommand.name == 'access') {
+        scrollToLink();
+      } else {
+        scrollPage();
+      }
     }
+  }
 
-    // Clear the command for next input
-    commandHistory.splice(1, 0, command);
-    command = "";
-    historyIndex = 0;
-    commandHistory[0] = "";
-    appendPrompt();
-}
+  // Clear the command for next input
+  commandHistory.splice(1, 0, command);
+  command = '';
+  historyIndex = 0;
+  commandHistory[0] = '';
+  appendPrompt();
+};
 
 /**
  * Renders things to the terminal on startup
  */
 const startTerminal = () => {
-    termwindow.append(`SCiPnet Data Network [version 0.0.1] ${span("status-success", "active")}\n\n`);
-    termwindow.append(span("status-red-alert", "WARNING. THE SCP FOUNDATION DATABASE IS CLASSIFIED! ACCESS BY UNAUTHORIZED PERSONNEL IS STRICTLY PROHIBITED! PERPETRATORS WILL BE TRACKED, LOCATED, AND DETAINED!\n\n"));
-
-    setLogin();
-
-    termwindow.append("Welcome to SCiPnet v0.0.1. For commands, type 'help'. To exit, type 'exit'. For more info, type 'manual'.\n")
-    appendPrompt();
-}
+  termwindow.append(
+    `SCiPnet Data Network [version 0.0.1] ${span(
+      'status-success',
+      'active'
+    )}\n\n`
+  );
+  termwindow.append(
+    span(
+      'status-red-alert',
+      'WARNING. THE SCP FOUNDATION DATABASE IS CLASSIFIED! ACCESS BY UNAUTHORIZED PERSONNEL IS STRICTLY PROHIBITED. PERPETRATORS WILL BE TRACKED, LOCATED, AND DETAINED.\n\n'
+    )
+  );
+
+  setLogin();
+
+  termwindow.append(
+    "Welcome to SCiPnet v0.0.1. For commands, type 'help'. To exit, type 'exit'. For more info, type 'manual'.\n"
```

Comment:
You may want to change the version to 0.1.0 because in semantic versioning the third number is patch changes (bug fixes, etc.) I recommend starting at 0.1.0 and for each breaking change increment the minor number. For pre-releases use 0.x.x-canary.0 and increment the "canary.0". For production releases, use version 1.x.x.

Reply:
Yea I didnt realize I was using the wrong version semantic. Because its a pre-release, instead of canary I'm just going to use alpha. If I remember correctly canary is for nightly builds.